packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/add-auth-request-adaptor.ts

```diff
@@ -0,0 +1,237 @@
+      (acc, it): AppIds => merge(acc, { [socialFederationKeyMap(it.provider, projectType)]: it.clientId }),
+      {} as AppIds,
+    ),
+  };
+  // adds "selectedParties" an "audiences" to the result which is a JSON string of the authProviders mapped to the client ids
+  identityPoolProviders(result, projectType);
+  return result;
+};
+
+// converts the password policy to the existing format
+const passwordPolicyMap = (pwPolicy?: CognitoPasswordPolicy): PasswordPolicyResult => {
+  if (!pwPolicy) return {};
+  return {
+    passwordPolicyMinLength: pwPolicy.minimumLength,
+    passwordPolicyCharacters: (pwPolicy.additionalConstraints || []).map(constraint => passwordConstraintMap[constraint]),
+  };
+};
+
+// converts admin queries config to existing format
+const adminQueriesMap = (adminQueries?: CognitoAdminQueries): AdminQueriesResult => {
+  return {
+    adminQueries: !!adminQueries,
+    adminQueryGroup: adminQueries?.permissions.groupName,
+  };
+};
+
+// converts mfa config to existing format
+const mfaMap = (mfaConfig: CognitoMFAConfiguration = { mode: 'OFF' }): MfaResult => {
+  if (mfaConfig.mode === 'OFF') {
+    return {
+      mfaConfiguration: 'OFF',
+    };
+  }
+  return {
+    mfaConfiguration: mfaConfig.mode,
+    mfaTypes: mfaConfig.mfaTypes.map(type => mfaTypeMap[type]),
+    smsAuthenticationMessage: mfaConfig.smsMessage,
+  };
+};
+
+// converts password recovery config to existing format
+const passwordRecoveryMap = (pwRecoveryConfig?: CognitoPasswordRecoveryConfiguration): PasswordRecoveryResult => {
+  switch (pwRecoveryConfig?.deliveryMethod) {
+    case 'SMS':
+      return {
+        smsVerificationMessage: pwRecoveryConfig?.smsMessage,
+        autoVerifiedAttributes: ['phone_number'],
+      };
+    case 'EMAIL':
+      return {
+        emailVerificationMessage: pwRecoveryConfig?.emailMessage,
+        emailVerificationSubject: pwRecoveryConfig?.emailSubject,
+        autoVerifiedAttributes: ['email'],
+      };
+    default:
+      return {
+        autoVerifiedAttributes: [],
+      };
+  }
+};
+
+const passwordConstraintMap: Record<CognitoPasswordConstraint, PasswordPolicy> = {
+  [CognitoPasswordConstraint.REQUIRE_LOWERCASE]: 'Requires Lowercase',
+  [CognitoPasswordConstraint.REQUIRE_DIGIT]: 'Requires Numbers',
+  [CognitoPasswordConstraint.REQUIRE_SYMBOL]: 'Requires Symbols',
+  [CognitoPasswordConstraint.REQUIRE_UPPERCASE]: 'Requires Uppercase',
+};
+
+const mfaTypeMap: Record<'SMS' | 'TOTP', 'SMS Text Message' | 'TOTP'> = {
+  SMS: 'SMS Text Message',
+  TOTP: 'TOTP',
+};
+
+const signinAttributeMap: Record<CognitoUserPoolSigninMethod, UsernameAttributes> = {
+  [CognitoUserPoolSigninMethod.USERNAME]: 'username',
+  [CognitoUserPoolSigninMethod.EMAIL]: 'email',
+  [CognitoUserPoolSigninMethod.PHONE_NUMBER]: 'phone_number',
+  [CognitoUserPoolSigninMethod.EMAIL_AND_PHONE_NUMBER]: 'email, phone_number',
+};
+
+// converts a word like tHiS to This
+const toTitleCase = (word: string): string =>
```

Comment:
We should use the library we already use for these case conversions.

Reply:
what lib do we use?